synapse/lib/ast.py

```diff
@@ -4972,6 +4972,11 @@ async def callfunc(self, runt, argdefs, args, kwargs):
 
                 except s_stormctrl.StormReturn as e:
                     return e.item
+                except s_stormctrl.StormCtrlFlow as e:
+                    if isinstance(e, s_stormctrl.StormLoopCtrl):
+                        mesg = f'{self.name} - Functions are not allowed to use unhandled loop control flow statements - {e._statement}'
```

Comment:
I'd phrase this more as "invalid" than "not allowed" :)

"Loop control flow statement used outside of a loop." or something.